packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -333,3 +327,79 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
         offlineDataJson[TimelineData.traceEventsKey] != null;
   }
 }
+
+class TimelineConfigurationsDialog extends StatelessWidget {
+  const TimelineConfigurationsDialog(this.controller);
+
+  static const dialogWidth = 300.0;
+
+  final TimelineController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
+      shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(defaultDialogRadius)),
+      child: Container(
+        width: dialogWidth,
+        padding: const EdgeInsets.all(defaultSpacing),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            ...headerInColumn(Theme.of(context).textTheme, 'Recorded Streams'),
+            ..._recordedStreams(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  List<Widget> _recordedStreams() {
+    final settings = <Widget>[];
+    for (final streamName in controller.recordedStreams.keys) {
+      final title = Text(streamName);
+      final checkbox = ValueListenableBuilder(
+        valueListenable: controller.recordedStreams[streamName],
+        builder: (context, enabled, _) {
+          return Checkbox(
```

Comment:
For many of these items, the user may not have a sense for what toggling them does. For the ones where we can come up with good, brief descriptions, perhaps include the description in-line and to the right pg the checkbox, w/ a subtle text style?

Reply:
That's a good idea. Added a TODO as I'm following up with @bkonyi to find accurate docs on these.

Reply:
OK; we may just need a good enough description to let the user know why they want want to turn these on.

I assume for many of them they really won't want to touch them. Some they will know what they do and want to enable them, like the network toggle. Not sure if there's a way to call out the two types well in the UI - have a regular and an advanced section?

Reply:
The observatory categorizes different sets like so:
![Screen Shot 2020-05-12 at 7 52 12 AM](https://user-images.githubusercontent.com/43759233/81707542-943dcb80-9425-11ea-94b0-fa7e1ef7970b.png)

Clicking one of these profiles enables a different set of timeline flags